docker-app/qfieldcloud/core/cron.py

```diff
@@ -35,3 +36,15 @@ def do(self):
         logger.info(
             f'Resend {len(invitation_emails)} previously failed invitation(s) to: {", ".join(invitation_emails)}'
         )
+
+
+class PurgeOldFileVersions(CronJobBase):
+    """Purges old version of files of all projects using the purge_old_file_versions
+    management command.
+    """
+
+    schedule = Schedule(runs_every_mins=60 * 24 * 7)
```

Comment:
Too low, I would say at least once per hour.

Reply:
I'm not exactly sure how cronjobs behave currently, but I guess this process can take some time in prod (ALL versions of ALL files of ALL projects)

Maybe it would be better to do that in a propre job queue ? It would allow to queue a project specific cleanup after file uploads, so that running it on all projects is only needed if we change the retention policy, and so that it's easier to scale if needed.

If we were to do that, would you suggest implementing this as a new Job class ?

Reply:
> I'm not exactly sure how cronjobs behave currently, but I guess this process can take some time in prod (ALL versions of ALL files of ALL projects)

This is only once when we enforce this.

> Maybe it would be better to do that in a propre job queue ? It would allow to queue a project specific cleanup after file uploads, so that running it on all projects is only needed if we change the retention policy, and so that it's easier to scale if needed.

I do agree in the long term, but not for now. I would say we just call this function right after each upload. I know it's not super scalable, but it simple and will work for while, whithout all the complexity of queued jobs.

> If we were to do that, would you suggest implementing this as a new Job class ?

The current jobs implementation needs to be tweaked a bit if we go to that direction. Also, they are executed sequentially per project, while the clean-up job can be ran in parallel.

Reply:
Ok did that:
- removed cron
- run purge_old_file_versions after uploads through API
- update tests accordingly